src/main/java/roomescape/domain/reservation/Payment.java

```diff
@@ -0,0 +1,106 @@
+package roomescape.domain.reservation;
+
+import jakarta.persistence.Column;
+import jakarta.persistence.Entity;
+import jakarta.persistence.FetchType;
+import jakarta.persistence.GeneratedValue;
+import jakarta.persistence.GenerationType;
+import jakarta.persistence.Id;
+import jakarta.persistence.JoinColumn;
+import jakarta.persistence.OneToOne;
+import java.math.BigDecimal;
+import java.util.Objects;
+import roomescape.domain.exception.DomainValidationException;
+
+@Entity
+public class Payment {
+
+    @Id
+    @GeneratedValue(strategy = GenerationType.IDENTITY)
+    private Long id;
+
+    @Column(nullable = false, unique = true)
+    private String paymentKey;
+
+    @Column(nullable = false, unique = true)
+    private String orderId;
+
+    @Column(nullable = false)
+    private BigDecimal amount;
+
+    @OneToOne(fetch = FetchType.LAZY)
+    @JoinColumn(nullable = false)
+    private Reservation reservation;
+
+    protected Payment() {
+    }
+
+    public Payment(String paymentKey, String orderId, BigDecimal amount, Reservation reservation) {
+        this(null, paymentKey, orderId, amount, reservation);
+    }
+
+    public Payment(Long id, String paymentKey, String orderId, BigDecimal amount, Reservation reservation) {
+        validate(paymentKey, orderId, amount, reservation);
+
+        this.id = id;
+        this.paymentKey = paymentKey;
+        this.orderId = orderId;
+        this.amount = amount;
+        this.reservation = reservation;
+    }
+
+    private void validate(String paymentKey, String orderId, BigDecimal amount, Reservation reservation) {
+        if (paymentKey == null || paymentKey.isBlank()) {
```

Comment:
요건 취향차이인 부분인데 `StringUtils` 를 사용하면 문자열 검사도 간결하게 할 수 있어요
 ```suggestion
        if (StringUtils.hasText(paymentKey)) {
```

다만 도메인 계층은 순수하게 사용하는게 좋다고하여
외부 라이브러리를 사용하지 않는게 좋다는 주장도 있습니다..!

하지만 이미 JPA 를 사용하여 Entity 설정을 위한 외부 라이브러리 의존도 있으니
저는 사용해도 괜찮다고 봐요 ㅎㅎ

꼭 사용할 필요는 없으니 참고만 해주세요!

Reply:
넵! JPA때문에 외부 라이브러리 의존이 있지만 최대한 순수하게 할려고 했어요.